### ISSUE TITLE
Modal: implement focus trap

### ISSUE THREAD

**Issue by bastaware** [Semantic-Org/Semantic-UI-React#2278]
Modal should not let focus leave the components inside the dialog. Also it should autofocus the first component in the dialog (if no autofocus was specified).

More on Modals and keyboard traps:
https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex

### Steps
1) Open any modal on the example page: https://react.semantic-ui.com/modules/modal
2) Press tab multiple times

### Expected Result
Focus should never enter the dimmed components below

### Actual Result
Focus does enter the dimmed components below

### Version
Current
Tested in Chrome Version 61.0.3163.100 (Official Build) (64-bit)

### Testcase
https://react.semantic-ui.com/modules/modal

**Comment by layershifter** [comment]
Yes, we should impliment focus trap on modal, [here](http://davidtheclark.github.io/react-aria-modal/demo/) is good example. However, it's a low priority enchantment. Contributions are welcome.

**Comment by fracmak** [comment]
I think this is a priority, reopening. Thoughts on adding this to the project dependencies? 

https://github.com/theKashey/react-focus-lock 

We've had luck using it in a few projects at my company.

**Comment by mjnaderi** [comment]
@fracmak Is there any plan to implement this? This is essential for addressing accessibility issues.

**Comment by medeaoblongata** [comment]
semantic-ui-react devs may find this helpful: https://github.com/WICG/inert/blob/master/explainer.md - note that `inert`, after some to-and-fro is becoming part of the HTML5 spec. See https://caniuse.com/?search=inert for progress on implementations.

`<dialog>` (the most correct semantic choice for HTML dialogs) has had better luck, with some support on all major browsers today. https://caniuse.com/?search=dialog

**Comment by ant100** [comment]
This is a very important accessibility feature, also found this other dependency that could help: https://github.com/focus-trap/focus-trap
